dask_bigquery/tests/test_core.py

```diff
@@ -0,0 +1,100 @@
+import random
+
+import pandas as pd
+import pytest
+from dask.dataframe.utils import assert_eq
+from distributed.utils_test import cluster_fixture  # noqa: F401
+from distributed.utils_test import client, loop  # noqa: F401
+from google.cloud import bigquery
+
+from dask_bigquery import read_gbq
+
+# These tests are run locally and assume the user is already athenticated.
+# It also assumes that the user has created a project called dask-bigquery.
+
+
+@pytest.fixture
+def df():
+    records = [
+        {
+            "name": random.choice(["fred", "wilma", "barney", "betty"]),
+            "number": random.randint(0, 100),
+            "idx": i,
+        }
+        for i in range(10)
+    ]
+
+    yield pd.DataFrame(records)
+
+
+@pytest.fixture
+def dataset(df):
+    "Push some data to BigQuery using pandas gbq"
+
+    try:
+        # delete data set if exists
+        bq_client = bigquery.Client()
+        bq_client.delete_dataset(
+            dataset="dask-bigquery.dataset_test",
+            delete_contents=True,
+        )
+        bq_client.close()
+    except:  # if data doesn't exisit continue is that Value Error?
+        pass
```

Comment:
If an error is raised when we call `delete_dataset`, then `bq_client.close()` won't ever be called. Let's put `bq_client.close()` in a `finally:` block to ensure it's called, even if an exception is raised. Something like:

```python
    try:
        # delete data set if exists
        bq_client = bigquery.Client()
        bq_client.delete_dataset(
            dataset="dask-bigquery.dataset_test",
            delete_contents=True,
        )
    except:  # if data doesn't exisit continue is that Value Error?
        pass
    finally:
        bq_client.close()
```

Reply:
Also, just to confirm, we can't use `bigquery.Client()` as a context manager, right?

Reply:
I was just double-checking this it might exists now. Haven't checked it  but will give it a try
https://github.com/googleapis/python-bigquery/pull/540

I had the feeling that this wasn't possible based on the code on the gist, but it seems most of this was drafted before the PR I mentioned was merged. But it seems only the storage API doesn't have the context manager nor a close method. 
So just for the test we could use the bigquery context manager. Will give it a try now.